lib/applications/resolvers.ts

```diff
@@ -713,3 +717,131 @@ export const generateApplicationsReport: Resolver = async (_, args, { prisma })
+  const csvApplications = applications.map(application => {
+    return {
+      ...application,
+      dateOfBirth: formatDate(application.dateOfBirth),
+      applicantName:
+        application.firstName +
+        (application.middleName
+          ? ` ${application.middleName} ${application.lastName}`
+          : ` ${application.lastName}`),
+      rcdPermitId: application.permit?.rcdPermitId,
+      homeAddress: `${application.addressLine1}, 
+      ${application.addressLine2 ? `${application.addressLine2},` : ''} ${application.city}, ${
+        application.province
+      }, ${application.postalCode}`,
+      userStatus: application.permit?.active ? 'Active' : 'Inactive',
+      guardianPOAName:
+        application.guardianFirstName +
+        (application.guardianMiddleName
+          ? ` ${application.guardianMiddleName} ${application.guardianLastName}`
+          : ` ${application.guardianLastName}`),
+      guardianPOAAdress: `${
+        application.guardianAddressLine1 ? `${application.guardianAddressLine1},` : ''
+      } ${application.guardianAddressLine2 ? `${application.guardianAddressLine2},` : ''} ${
+        application.guardianCity ? `${application.guardianCity},` : ''
+      } ${application.guardianProvince ? `${application.guardianProvince},` : ''} ${
+        application.guardianPostalCode ?? ''
+      }`,
+    };
+  });
+
+  const csvHeaders = [];
+
+  if (columnsSet.has(PermitHoldersReportColumn.UserId)) {
+    csvHeaders.push({ id: 'rcdUserId', title: 'User ID' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.ApplicantName)) {
+    csvHeaders.push({ id: 'applicantName', title: 'Applicant Name' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.ApplicantDateOfBirth)) {
+    csvHeaders.push({ id: 'dateOfBirth', title: 'Applicant DoB' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.HomeAddress)) {
+    csvHeaders.push({ id: 'homeAddress', title: 'Home Address' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.Email)) {
+    csvHeaders.push({ id: 'email', title: 'Email' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.PhoneNumber)) {
+    csvHeaders.push({ id: 'phone', title: 'Phone Number' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.GuardianPoaName)) {
+    csvHeaders.push({ id: 'guardianPoaName', title: 'Guardian/POA Name' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.GuardianPoaRelation)) {
+    csvHeaders.push({ id: 'guardianRelationship', title: 'Guardian/POA Relation' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.GuardianPoaAddress)) {
+    csvHeaders.push({ id: 'guardianPOAAdress', title: 'Guardian/POA Address' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.RecentAppNumber)) {
+    csvHeaders.push({ id: 'rcdPermitId', title: 'Recent APP Number' });
+  }
+  if (columnsSet.has(PermitHoldersReportColumn.RecentAppType)) {
+    csvHeaders.push({ id: 'permitType', title: 'Recent APP Type' });
+  }
```

Comment:
For most recent APP type, it should be on permits, not applications. Our DB currently doesn't support this, but I will add a ticket to the backlog to address this. For now, could you please use the `permit_type` column on the most recent application and add a TODO comment to update this? Thanks :)

Reply:
Addressed with the following:

![image](https://user-images.githubusercontent.com/51224641/147160816-cc030d16-3b29-4359-b7fc-72501b6105fe.png)